diff --git a/Classes/ViewHelpers/Mark/ContentAreaViewHelper.php b/Classes/ViewHelpers/Mark/ContentAreaViewHelper.php
index db29add..e526e47 100644
--- a/Classes/ViewHelpers/Mark/ContentAreaViewHelper.php
+++ b/Classes/ViewHelpers/Mark/ContentAreaViewHelper.php
@@ -11,6 +11,7 @@ namespace TYPO3\CMS\VisualEditor\ViewHelpers\Mark;
 use Exception;
 use B13\Container\Tca\Registry;
 use InvalidArgumentException;
+use TYPO3\CMS\Core\Page\ContentAreaClosure;
 use Psr\Http\Message\ServerRequestInterface;
 use TYPO3\CMS\Backend\Utility\BackendUtility;
 use TYPO3\CMS\Core\EventDispatcher\EventDispatcher;
@@ -102,6 +103,9 @@ final class ContentAreaViewHelper extends AbstractViewHelper
         /** @var PageInformation $pageInformation */
         $pageInformation = $request->getAttribute('frontend.page.information');
         $pageLayout = $pageInformation->getPageLayout() ?? throw new InvalidArgumentException('PageLayout is not available', 1772464283);
         foreach ($pageLayout->getContentAreas() as $contentArea) {
+            if ($contentArea instanceof ContentAreaClosure) {
+                $contentArea = $contentArea->instantiate();
+            }
             if ($this->typo3Version->getMajorVersion() >= 14) {
                 $contentAreaColPos = $contentArea->getColPos();
                 $name = $contentArea->getName();
