diff --git a/Classes/Module/ModuleProvider.php b/Classes/Module/ModuleProvider.php
index a44e2c7b2..5f26b6857 100644
--- a/Classes/Module/ModuleProvider.php
+++ b/Classes/Module/ModuleProvider.php
@@ -280,6 +280,7 @@ readonly class ModuleProvider
             }
             // Skip submodules that depend on their own submodules if they don't have any accessible ones
             if (($subModule->getAppearance()['dependsOnSubmodules'] ?? false)
+                && !$subModule->hasSubmoduleOverview()
                 && !$this->hasAccessibleSubModules($subModule, $hideModules, $user, $respectWorkspaceRestrictions, $checkRenderInModuleMenu)
             ) {
                 continue;
